pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -578,7 +585,32 @@ func (pl *dynamicResources) PreFilter(ctx context.Context, state *framework.Cycl
 			if err != nil {
 				return nil, statusError(logger, err)
 			}
-			s.availableOnNodes[index] = nodeSelector
+			s.informationsForClaim[index].availableOnNode = nodeSelector
+		}
+		if claim.Status.Allocation == nil &&
+			claim.Spec.AllocationMode == resourcev1alpha2.AllocationModeWaitForFirstConsumer {
+			// The ResourceClass might have a node filter. This is
+			// useful for trimming the initial set of potential
+			// nodes before we ask the driver(s) for information
+			// about the specific pod.
+			class, err := pl.classLister.Get(claim.Spec.ResourceClassName)
+			if err != nil {
+				// If the class does not exist, then allocation cannot proceed.
+				return nil, statusError(logger, fmt.Errorf("look up resource class: %v", err))
```

Comment:
What's the possible scenario that the class doesn't exist? If ResourceClass should be created by users, then can we return UnschedulableAndUnresolvable instead of Error if it's not found error? If we return error here, the Pod will come back to the scheduling cycle soon. (the scheduling queue will enqueue this Pod to backoffQ instead of unsched pod pool.)

Reply:
> What's the possible scenario that the class doesn't exist?

It's a user error. I think treating this as an unexpected error is better than trying to handle this gracefully when "handle gracefully" implies adding more code.

> can we return UnschedulableAndUnresolvable instead of Error if it's not-found error?

I suppose we could, but then we would also have to register cluster events for "ResourceClass created" to get the pod out of the unschedulable queue, right?

Not sure whether that's worth it.

Reply:
> we would also have to register cluster events for "ResourceClass created" to get the pod out of the unschedulable queue, right?

Yes, right.

> Not sure whether that's worth it.

It should be worthwhile because otherwise, such Pods keep being scheduled → rejected here → pushed to backoffQ/activeQ →..., and waste scheduling cycles until ResourceClass is created.

So, in my eyes, the change would be just to handle not-found error here + to register ResourceClass created event in EventsToRegister, which doesn't add much complexity to the code.

Reply:
Can we do that in a follow-up PR?

Reply:
The code changes will also needs tests and those changes are unrelated to the purpose of this PR.

Reply:
> Can we do that in a follow-up PR?

Yep, makes sense.

Reply:
=> https://github.com/kubernetes/enhancements/issues/3063